### PR TITLE
Logging level dictated by env var

### PR DESCRIPTION
**Pull request by abottoms-coder**
### Summary of Changes
Closes issue #1036 

**Acceptance Criteria:** 
_Create a list of functional outcomes that must be achieved to complete this issue_

- [x] Django log level can be set via environment variable
- [x] Django logging handlers are used for all cases that adjust log level
- [x] Testing Checklist has been run and all tests pass
- [x] reviewed and tested by @ADPennington 

### How to Test

1. Add `LOGGING_LEVEL=DEBUG` to your `.env` file
2. Spin up your containers:
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```

3. Find your container id then get log feedback in real-time
```
docker ps|grep tdp
docker logs -f <container id>
```
4. Open http://localhost:3000/ and sign in.
5. With a terminal open to view the logs in real time, your login should've produced debug-level logging in real time.
6. Test steps should be captured in the demo GIF(s) and/or screenshots below.

> *Demo GIF(s) and screenshots for testing procedure*

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [x] Look up the acceptance criteria in the related issue; paste ACs below in checklist format.
+ [x] Check against the criteria:

As Product Owner, @lfrohlich will decide if ACs are met.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)
  + [x] Backend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

+ [x] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Does this PR change any linting or CI settings?

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*
+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with @iamjolly and @ttran-hub using Accessibility Insights reveal any errors introduced in this PR?
    + [See the full Accessibility Assessment plan here.](https://github.com/HHS/TANF-app/blob/main/docs/Technical-Documentation/how-government-will-test-a11y.md)

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful deployment by assigning a 'Deploy with CircleCI - {env}' label*

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

+ [x] If this PR introduces backend code, is that code documented both inline and overall?
+ [ ] If this PR introduces frontend code, is that code documented both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] Does the OWASP Scan pass on CircleCI?
+ [x] Do manual code review and manual testing detect any security issues?

### Deliverable 8: Context

> * Performance Standard(s): Code must be understandable and contextualized for the reviewers possess the knowledge and background necessary for analysis and constructive criticism to take place.

> * Acceptable Quality Level: Code submitted in the pull request has context.*
+ [x] Does this pull request contain sufficient inline comments providing relevant context for code snippets?
+ [x] Is the code understandable and lucid?
+ [x] Does this pull request provide background for why coding decisions were made?
+ [x] Can you as a reviewer explain and take ownership of these elements presented in this code review?

### CODE REVIEW

File: tdrs-backend/tdpservice/settings/common.py [hunk 1]
```diff
@@ -198,30 +200,30 @@ class Common(Configuration):
...
             },
             "console": {
-                "level": "DEBUG",
+                "level": LOGGING_LEVEL,
                 "class": "logging.StreamHandler",
                 "formatter": "simple",
             },
             "mail_admins": {
-                "level": "ERROR",
+                "level": LOGGING_LEVEL,
```
Comment (jtwillis92) [inline tdrs-backend/tdpservice/settings/common.py:213]: We're not actually using the admin email functionality ([the emails specified in our settings are invalid](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/tdrs-backend/tdpservice/settings/common.py#L90))

However, if we were I'd think we wouldn't want to send `INFO` level logs to those emails. I'd suggest either removing this handler for simplicity or just keeping this one at `ERROR`

File: tdrs-backend/tdpservice/users/api/authorization_check.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 """Check if user is authorized."""
-import logging
 
+from django.conf import settings
```
Comment (jtwillis92) [inline tdrs-backend/tdpservice/users/api/authorization_check.py:3]: this import looks unused now

File: tdrs-backend/tdpservice/users/serializers.py [hunk 3]
```diff
@@ -3,13 +3,11 @@
 import logging
 from django.contrib.auth.models import Group, Permission
 from rest_framework import serializers
-
+from django.conf import settings
```
Comment (jtwillis92) [inline tdrs-backend/tdpservice/users/serializers.py:6]: unused as well

File: tdrs-backend/tdpservice/settings/common.py [hunk 4]
```diff
@@ -179,6 +179,8 @@ class Common(Configuration):
     ]
 
     # Logging
+    # set level as 'INFO' if env var is not set
+    LOGGING_LEVEL = os.getenv('LOGGING_LEVEL', 'INFO')
```
Comment (jtwillis92) [inline tdrs-backend/tdpservice/settings/common.py:183]: I would also suggest defining this in `.env.example` so we have a full list of available variables.

**Review by jtwillis92 — APPROVED** [review]
LGTM, pulled this branch down locally and tested changing the logging level.

One note for the testing steps, you can just do `docker-compose logs -f web` to get the backend container logs.

**Review by jorgegonzalez — APPROVED** [review]
Tested this locally, thanks, this is a nice quality of life change 👍

**Comment by ADPennington** [comment]
@abottoms-coder can you look into the branch conflict here?

**Review by ADPennington — APPROVED** [review]
lgtm 👍🏾

**Comment by abottoms-coder** [comment]
> lgtm 👍🏾 but please look into backend owasp scan. this was passing prior to the most recent merge. @abottoms-coder

It was a simple timeout on the container. Re-running

**Comment by abottoms-coder** [comment]
> > lgtm 👍🏾 but please look into backend owasp scan. this was passing prior to the most recent merge. @abottoms-coder
> 
> It was a simple timeout on the container. Re-running

Timeout is persistent even in local environment. Will reach out to @jtwillis92 for further review

```2021-12-07T17:32:58:INFO:localstack.utils.analytics.profiler: Execution of "start_api_services" took 116889.34516906738ms
2021-12-07 17:33:00,884 INFO exited: infra (exit status 2; not expected)
2021-12-07 17:33:00,886 INFO spawned: 'infra' with pid 106
ERROR: 'NoneType' object is not subscriptable
(. .venv/bin/activate; exec bin/localstack start --host)
make: *** [Makefile:54: infra] Error 1
2021-12-07 17:33:01,899 INFO success: infra entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
Waiting for all LocalStack services to be ready
Starting local dev environment. CTRL-C to quit.

LocalStack version: 0.12.9
LocalStack build date: 2021-03-29
LocalStack build git hash: 2dd4672

Starting edge router (https port 4566)...
Starting mock S3 service on http port 4566 ...
2021-12-07T17:33:07:INFO:localstack.utils.analytics.profiler: Execution of "load_plugin_from_path" took 5430.908918380737ms
2021-12-07T17:33:07:INFO:localstack.utils.analytics.profiler: Execution of "load_plugins" took 5432.187080383301ms
2021-12-07T17:33:08:INFO:localstack.multiserver: Starting multi API server process on port 44883
[2021-12-07 17:33:08 +0000] [107] [INFO] Running on https://0.0.0.0:4566 (CTRL + C to quit)
2021-12-07T17:33:08:INFO:hypercorn.error: Running on https://0.0.0.0:4566 (CTRL + C to quit)
[2021-12-07 17:33:08 +0000] [107] [INFO] Running on http://0.0.0.0:44883 (CTRL + C to quit)
2021-12-07T17:33:08:INFO:hypercorn.error: Running on http://0.0.0.0:44883 (CTRL + C to quit)
Waiting for all LocalStack services to be ready
Waiting for all LocalStack services to be ready
2021-12-07T17:33:19:WARNING:localstack.services.plugins: Service "s3" not yet available, retrying...
Waiting for all LocalStack services to be ready
2021-12-07T17:33:27:WARNING:localstack.services.plugins: Service "s3" not yet available, retrying...
Waiting for all LocalStack services to be ready
2021-12-07T17:33:39:WARNING:localstack.services.plugins: Service "s3" not yet available, retrying...
Waiting for all LocalStack services to be ready
Waiting for all LocalStack services to be ready
2021-12-07T17:33:47:WARNING:localstack.services.plugins: Service "s3" not yet available, retrying...
Waiting for all LocalStack services to be ready
```
Ad naseum.

**Comment by abottoms-coder** [comment]
@ADPennington 
When you get a chance, please re-review this ticket as I found the underlying culprit: I had not wired up circleCI to accept the environment variable which is why I was only successful locally.

**Review by ADPennington — APPROVED** [review]
👍🏾

**Pull request merged**